tests/test_utils.py

```diff
@@ -4,127 +4,104 @@
+def double_op_call_string(name: str, qb1: int, qb2: int) -> str:
+    return f"call void @__quantum__qis__{name}__body({_qubit_string(qb1)}, {_qubit_string(qb2)})"
 
 
-def adj_op_call_string(name: str, qb: int, static_alloc=True) -> str:
-    return f"call void @__quantum__qis__{name}__adj({_qubit_string(qb, static_alloc)})"
+def rotation_call_string(name: str, theta: float, qb: int) -> str:
+    return f"call void @__quantum__qis__{name}__body(double {theta:#e}, {_qubit_string(qb)})"
 
 
-def double_op_call_string(name: str, qb1: int, qb2: int, static_alloc=True) -> str:
-    return f"call void @__quantum__qis__{name}__body({_qubit_string(qb1, static_alloc)}, {_qubit_string(qb2, static_alloc)})"
-
-
-def rotation_call_string(name: str, theta: float, qb: int, static_alloc=True) -> str:
-    return f"call void @__quantum__qis__{name}__body(double {theta:#e}, {_qubit_string(qb, static_alloc)})"
-
-
-def measure_call_string(
-    name: str, res: str, qb: int, static_qubit_alloc=True, static_result_alloc=True
-) -> str:
-    if static_result_alloc:
-        return f"call void @__quantum__qis__{name}__body({_qubit_string(qb, static_qubit_alloc)}, {_result_string(res, static_result_alloc)})"
-    else:
-        return f"%result{res} = call %Result* @__quantum__qis__{name}__body({_qubit_string(qb, static_qubit_alloc)})"
+def measure_call_string(name: str, res: str, qb: int) -> str:
+    return f"call void @__quantum__qis__{name}__body({_qubit_string(qb)}, {_result_string(res)})"
 
 
 def equal(var: str, res: str):
     return f"%{var} = call i1 @__quantum__qis__read_result__body({_result_string(res)})"
 
 
-def generic_op_call_string(name: str, qbs: List[int], static_alloc=True) -> str:
-    args = ", ".join(_qubit_string(qb, static_alloc) for qb in qbs)
+def generic_op_call_string(name: str, qbs: List[int]) -> str:
+    args = ", ".join(_qubit_string(qb) for qb in qbs)
     return f"call void @__quantum__qis__{name}__body({args})"
 
 
 def return_string() -> str:
     return "ret void"
 
 
-def array_start_record_output_string() -> str:
-    return f"call void @__quantum__rt__array_start_record_output()"
+def array_record_output_string(num_elements: int) -> str:
+    return (
+        f"call void @__quantum__rt__array_record_output(i64 {num_elements}, i8* null)"
+    )
 
 
-def array_end_record_output_string() -> str:
-    return f"call void @__quantum__rt__array_end_record_output()"
+def result_record_output_string(res: str) -> str:
+    return f"call void @__quantum__rt__result_record_output({_result_string(res)}, i8* null)"
 
 
-def result_record_output_string(res: str, static_alloc=True) -> str:
-    return f"call void @__quantum__rt__result_record_output({_result_string(res, static_alloc)})"
+# Returns the method body with:
+# - leading spaces trimmed
+# - first label skipped
+# - signature and closing braces removed
+def find_function(qir: List[str]) -> List[str]:
```

Comment:
It's not clear from the name and comment that this is finding the *entry point* function.